src/server/detail/save_stages_controller.cc

```diff
@@ -151,10 +151,11 @@ SaveStagesController::SaveStagesController(SaveStagesInputs&& inputs)
 SaveStagesController::~SaveStagesController() {
 }
 
-SaveInfo SaveStagesController::Save() {
+std::optional<SaveInfo> SaveStagesController::InitResourcesAndStart() {
   if (auto err = BuildFullPath(); err) {
     shared_err_ = err;
-    return GetSaveInfo();
+    SaveInfo save = GetSaveInfo();
+    return std::optional<SaveInfo>(std::move(save));
```

Comment:
On my defense, I was experimenting with a false positive from you complete me. Basically I had it like this and I got a warning on Vim so I changed it because I was curious if the warning would go away (it didn't) and I left forgotten as is :eyes: